src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -971,16 +960,224 @@ void CmdSketcherViewSection::activated(int iMsg)
 
 bool CmdSketcherViewSection::isActive()
 {
-    Gui::Document *doc = getActiveGuiDocument();
-    if (doc) {
-        // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        SketcherGui::ViewProviderSketch* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
-        if (vp /*&& vp->getSketchMode() == ViewProviderSketch::STATUS_NONE*/)
-            return true;
+    return isSketchInEdit(getActiveGuiDocument());
+}
+
+/* Grid tool */
+class GridSpaceAction : public QWidgetAction
+{
+public:
+    GridSpaceAction(QObject* parent) : QWidgetAction(parent) {
     }
+
+    void updateWidget() {
+
+        auto* sketchView = getView();
+
+        if(sketchView) {
+
+            auto updateCheckBox = [this](QCheckBox * checkbox, App::PropertyBool & property) {
+                auto checked = checkbox->checkState() == Qt::Checked;
+                auto propvalue = property.getValue();
+
+                if( propvalue != checked ) {
+                    checkbox->blockSignals(true);
+                    checkbox->setChecked(propvalue);
+                    checkbox->blockSignals(false);
+                }
+            };
+
+            updateCheckBox(gridSnap, sketchView->GridSnap);
+            updateCheckBox(gridAutoSpacing, sketchView->GridAuto);
+
+            auto autospacing = gridAutoSpacing->checkState() == Qt::Checked;
+
+            gridSizeBox->setEnabled(!autospacing);
+
+            gridSizeBox->setValue(sketchView->GridSize.getValue());
+        }
+    }
+
+protected:
+    QWidget* createWidget(QWidget* parent) override
+    {
+        gridSnap = new QCheckBox(QApplication::translate("GridSpaceAction", "Grid Snap"));
+        gridSnap->setToolTip(QApplication::translate("CmdSketcherCompGrid", "New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid spacing to a grid line to snap."));
+        gridSnap->setStatusTip(gridSnap->toolTip());
+
+        gridAutoSpacing = new QCheckBox(QApplication::translate("GridSpaceAction", "Grid Auto Spacing"));
+        gridAutoSpacing->setToolTip(QApplication::translate("CmdSketcherCompGrid", "Resize grid automatically depending on zoom."));
+        gridAutoSpacing->setStatusTip(gridAutoSpacing->toolTip());
+
+        auto* sizeLabel = new QLabel(QApplication::translate("GridSpaceAction", "Spacing"));
+        gridSizeBox = new Gui::QuantitySpinBox();
+        gridSizeBox->setProperty("unit", QVariant(QStringLiteral("mm")));

```

Comment:
Wait. I'm not sure but IIRC, actually it does adapt to user schema when displayed.

Reply:
I did not have time to look into it yet. I was hopping to find something more expressive, line "LengthUnit" instead of having to hardcode "mm". I still have it in the todo list. It was a difficult day.

Reply:
No problem. I just checked with some Python :
```
qsb = FreeCADGui.UiLoader().createWidget('Gui::QuantitySpinBox')
qsb.setProperty('unit', 'in')
qsb.show()
```
And here it shows in 'mm' conforming to my user schema.
If you have an imperial schema, you can make same test replacing 'in' with 'mm' and it should also work. ;)